projects/control-service/projects/pipelines_control_service/src/main/java/com/vmware/taurus/service/execution/JobExecutionUtil.java

```diff
@@ -0,0 +1,121 @@
+/*
+ * Copyright 2021 VMware, Inc.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.vmware.taurus.service.execution;
+
+import java.time.Instant;
+import java.util.Map;
+
+import com.google.gson.Gson;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.lang3.StringUtils;
+
+import com.vmware.taurus.service.KubernetesService;
+import com.vmware.taurus.service.model.ExecutionStatus;
+import com.vmware.taurus.service.model.TerminationMessage;
+import com.vmware.taurus.service.model.TerminationStatus;
+
+@Slf4j
+public class JobExecutionUtil {
+
+   static final String TERMINATION_MESSAGE_ATTRIBUTE_STATUS = "status";
+   static final String TERMINATION_MESSAGE_ATTRIBUTE_VDK_VERSION = "vdk_version";
+
+   public static TerminationMessage getTerminationMessage(String terminationMessage) {
+      String vdkVersion = "";
+      String status = "";
+
+      if (!StringUtils.isEmpty(terminationMessage)) {
+         try {
+            Map<String, String> obj = new Gson().fromJson(terminationMessage, Map.class);
+            status = obj.get(TERMINATION_MESSAGE_ATTRIBUTE_STATUS);
+            vdkVersion = obj.getOrDefault(TERMINATION_MESSAGE_ATTRIBUTE_VDK_VERSION, "");
+         } catch (com.google.gson.JsonSyntaxException ex) {
+            // Fallback to the old plain text format
+            status = terminationMessage;
+         }
+      }
+
+      return TerminationMessage.builder()
+            .terminationStatus(getTerminationStatus(status))
+            .vdkVersion(vdkVersion)
+            .build();
+   }
+
+   static ExecutionStatus getJobExecutionStatus(KubernetesService.JobExecution jobExecution, TerminationStatus terminationStatus) {
+
+      if (isJobExecutionSkipped(terminationStatus)) {
+         return ExecutionStatus.SKIPPED;
+      } else if (isJobExecutionFailed(terminationStatus)) {
+         return ExecutionStatus.FAILED;
+      }
+
+      var jobStatus = jobExecution.getStatus();
+      switch (jobStatus) {
+         case RUNNING:
+            return ExecutionStatus.RUNNING;
+         case FAILED:
+            return ExecutionStatus.FAILED;
+         case FINISHED:
+            return ExecutionStatus.FINISHED;
+         case SKIPPED:
+            return ExecutionStatus.SKIPPED;
+         case CANCELLED:
+            return ExecutionStatus.CANCELLED;
+         default:
+            log.warn("Unexpected job status: '" + jobStatus + "' in JobExecutionStatus.Status.");
+            return null;
+      }
+   }
+
+   static String getExecutionId(String jobName) {
+      return String.format("%s-%s", jobName, Instant.now().getEpochSecond());
+   }
+
+   static String buildStartedByAnnotationValue(JobExecutionService.ExecutionType executionType, String startedBy) {
+      return executionType.getValue() + "/" + startedBy;
```

Comment:
we do not need to include execution type here if there is a separate field for it.